lib/pages/ui/nps_rating_bar.dart

```diff
@@ -2,28 +2,42 @@ import 'package:flutter/cupertino.dart';
 import 'package:flutter/material.dart';
 
 const String RATING_TYPE_NPS = "nps";
+const int DEFAULT_SELECTED_RATE = 4;
 
 class NpsRatingBar extends StatefulWidget {
-  final int counter;
+  final List<String> ids;
   final String minTitle;
   final String maxTitle;
+  final Function(Map<String, String>) onRatingListener;
 
-  NpsRatingBar({Key key, this.counter, this.minTitle, this.maxTitle})
+  NpsRatingBar(
+      {Key key,
+      this.ids,
+      this.minTitle,
+      this.maxTitle,
+      @required this.onRatingListener})
       : super(key: key);
 
   @override
   _NpsRatingBarState createState() => _NpsRatingBarState();
 }
 
 class _NpsRatingBarState extends State<NpsRatingBar> {
-  int _selectedRate = 4;
+  int _selectedRate = DEFAULT_SELECTED_RATE;
 
   void _onRateSelected(int rate) {
     setState(() {
       _selectedRate = rate;
+      widget.onRatingListener.call({widget.ids[rate]: ""});
```

Comment:
just an improvement: we should put the value that could affect to view state into `setState`, other values can be outside 😄